src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java

```diff
@@ -0,0 +1,262 @@
+package org.openmrs.contrib.qaframework.automation;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.reference.page.ActiveVisitsPage;
+import org.openmrs.reference.page.AddOrEditAllergyPage;
+import org.openmrs.reference.page.AllergyPage;
+import org.openmrs.reference.page.AttachmentsPage;
+import org.openmrs.reference.page.ConditionPage;
+import org.openmrs.reference.page.ConditionsPage;
+import org.openmrs.reference.page.PatientVisitsDashboardPage;
+import org.openmrs.reference.page.RequestAppointmentPage;
+import org.openmrs.reference.page.VisitNotePage;
+import org.openmrs.uitestframework.test.TestData;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class ClinicalVisitSteps extends Steps {
+
+	private static final String DIAGNOSIS_PRIMARY = "Cancer";
+	private static final String DIAGNOSIS_SECONDARY = "Malaria";
+	private static final String APPOINTMENT_TYPE = "Surgery";
+	private static final String DRUG_NAME = "Penicillins";
+	private static final String REACTION = "Headache";
+	private static final String CONDITION = "Malaria";
+	private ActiveVisitsPage activeVisitsPage;
+	private VisitNotePage visitNotePage;
+	private AllergyPage allergyPage;
+	private AddOrEditAllergyPage addOrEditAllergyPage;
+	private ConditionPage conditionPage;
+	private ConditionsPage conditionsPage;
+	private AttachmentsPage attachmentsPage;
+	private RequestAppointmentPage requestAppointmentPage;
+	private TestData.PatientInfo testPatient;
+
+	@Before(RunTest.HOOK.SELENIUM_CLINICAL_VISIT)
+	public void visitHomePage() {
+		testPatient = createTestPatient();
+		initiateWithLogin();
+		new TestData.TestVisit(testPatient.uuid, TestData.getAVisitType(),
+				getLocationUuid(homePage)).create();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_CLINICAL_VISIT)
+	public void destroy() {
+		deletePatient(testPatient);
+		quit();
+	}
+
+	@Given("a user clicks on active visits link from home page")
+	public void launchPatientDashboardActiveVisits() {
+		activeVisitsPage = homePage.goToActiveVisitsSearch();
+	}
+
+	@When("a user selects a patient from active patient list")
+	public void searchActivePatient() {
+		activeVisitsPage.search(testPatient.identifier);
+		dashboardPage = activeVisitsPage
+				.goToPatientDashboardOfLastActiveVisit();
+	}
+
+	@Then("the system loads Patient dashboard page")
+	public void launchPatientDashboardPage() {
+		assertNotNull(dashboardPage.getActiveVisitList());
+	}
+
+	// User story: Complete visit note
```

Comment:
Do you need this comment here

Reply:
> Do you need this comment here

Greta comment!
The fact that the workflow has many user stories, I though it wise that putting meaningful comment in the steps definition is a good deal in making easy code review easier and as well documenting the each user stories in the steps definition